Feature 0.8 (#459)
MicrosoftDocs/quantum-docs-pr: articles/language/expressions.md

All arrays in Q# are zero-based.
That is, the first element of an array `a` is always `a[0]`.

> :new: Starting with our 0.8 release, we are supporting contextual expressions for range slicing. 

Reviewer: ```suggestion
:new: Starting with our 0.8 release, we support contextual expressions for range slicing. 
```

> In particular, range start and end values may be omitted in the context of a range slicing expression. 

Reviewer 2: You should say something here about using ellipsis, "...", instead of just leaving the value off.

Author: Should I? I had it listed as example, but right now it has no purpose whatsoever, so it seems that talking about it would make much more sense when we have the multidimensional case. I'll add back the example though.

> In that case, the compiler will make a best guess as to what the intended delimiters are for the range. 

Reviewer 2: ```suggestion
In that case, the compiler will apply the following rules to infer the intended delimiters for the range. 
```

> That means that if for example the range start value is omitted, 

Reviewer 2: ```suggestion
For example, if the range start value is omitted, 
```

> then the intended start value 

Reviewer 2: ```suggestion
then the inferred start value 
```

> - is assumed to be zero if no step is specified or the specified step is positive, and 

Reviewer 2: ```suggestion
- is zero if no step is specified or the specified step is positive, and 
```

> - is assumed to be the length of sliced array minus one if the specified step is negative. 

Reviewer 2: ```suggestion
- is the length of sliced array minus one if the specified step is negative. 
```

If the range end value is omitted, 

> then the intended end value 

Reviewer 2: ```suggestion
then the inferred end value 
```

> - is assumed to be the length of sliced array minus one if no step is specified or the specified step is positive, and 

Reviewer 2: ```suggestion
- is the length of sliced array minus one if no step is specified or the specified step is positive, and 
```

> - is assumed to be zero if the specified step is negative. 

Reviewer 2: ```suggestion
- is zero if the specified step is negative. 
```
